runtime/compiler/control/CompilationThread.cpp

```diff
@@ -9256,6 +9283,36 @@ TR::CompilationInfoPerThreadBase::wrappedCompile(J9PortLibrary *portLib, void *
                reloRuntime,
                &target);
 
+#if defined(J9VM_OPT_JITSERVER)
+if (that->_methodBeingCompiled->isOutOfProcessCompReq())
+   {
+   // use the default code cache
+   compiler->getOptions()->setCodeCacheKind(TR::CodeCacheKind::DEFAULT_CC);
+   }
+else
+#endif
+   {
+   TR::SimpleRegex *regex = compiler->getOptions()->getTransientClassRegex();
+   if (regex)
+      {
+      J9Method *method = details.getMethod();
+      J9Class *clazz = J9_CLASS_FROM_METHOD(method);
+      J9UTF8 *className = J9ROMCLASS_CLASSNAME(clazz->romClass);
+      // TR::SimpleRegex::match needs a NULL-terminated string
+      size_t classNameLength = J9UTF8_LENGTH(className);
+      char *name = (char*)compiler->trMemory()->allocateMemory(classNameLength+1, stackAlloc);
+      strncpy(name, (char*)J9UTF8_DATA(className), classNameLength);
+      name[classNameLength] = '\0';
+      J9ClassLoader *clazzLoader = clazz->classLoader;
+      TR::CompilationInfo *compInfo = that->getCompilationInfo();
+      bool classLoaderMarkedTransient = compInfo->isClassLoaderMarkedTransient(clazzLoader);
+      if (classLoaderMarkedTransient || TR::SimpleRegex::match(regex, name))
+         {
+         compiler->getOptions()->setCodeCacheKind(TR::CodeCacheKind::TRANSIENT_CODE_CC);
+         if (!classLoaderMarkedTransient) compInfo->markClassLoaderTransient(clazzLoader);
```

Comment:
Following OpenJ9 coding conventions, the body of the if statement should go on a separate line.